otter/convergence/steps.py

```diff
@@ -70,12 +102,38 @@ def got_name(random_name):
                 'POST',
                 'servers',
                 data=thaw(server_config),
-                success_pred=has_code(202))
+                success_pred=has_code(202),
+                reauth_codes=(401,))
 
         def report_success(result):
             return StepResult.SUCCESS, []
 
         def report_failure(result):
+            """
```

Comment:
I've moved it to another function, although currently just a private function.  This definitely decreases the complexity of the `report_failure` function, thanks  :)

I think the only part that may be de-duplicatable between this function and `UpstreamError` might be in `_try_json_message`, since UpstreamError attempts to just get a nice message out of something that looks like `{'key': {'message': '...'}}`, and the parsing of nova errors is checking for specific messages and specific keys.